hypershift-operator/controllers/hostedcluster/metrics/metrics.go

```diff
@@ -456,30 +463,41 @@ func (c *hostedClustersMetricsCollector) Collect(ch chan<- prometheus.Metric) {
 			}
 
 			// etcdManualInterventionRequiredMetric
+			// clusterSizeOverrideMetric
 			{
 				metricLabels := make(map[string]string, 0)
-				if hcluster.Spec.Platform.Type == hyperv1.AWSPlatform && hcluster.Spec.Platform.AWS.ResourceTags != nil {
+				if hcluster.Spec.Platform.Type == hyperv1beta1.AWSPlatform && hcluster.Spec.Platform.AWS.ResourceTags != nil {
 					for _, resourceTag := range hcluster.Spec.Platform.AWS.ResourceTags {
 						switch resourceTag.Key {
 						case "api.openshift.com/environment":
 							metricLabels["environment"] = resourceTag.Value
 						case "api.openshift.com/id":
-							metricLabels["id"] = resourceTag.Value
+							metricLabels["internal_id"] = resourceTag.Value
 						case "red-hat-clustertype":
 							metricLabels["cluster_type"] = resourceTag.Value
 						}
 					}
 				}
 
 				if metricLabels["cluster_type"] == "rosa" {
-					etcdRecoveryActiveCondition := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.EtcdRecoveryActive))
-					if etcdRecoveryActiveCondition != nil && etcdRecoveryActiveCondition.Status == metav1.ConditionFalse && etcdRecoveryActiveCondition.Reason == hyperv1.EtcdRecoveryJobFailedReason {
+					etcdRecoveryActiveCondition := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1beta1.EtcdRecoveryActive))
+					if etcdRecoveryActiveCondition != nil && etcdRecoveryActiveCondition.Status == metav1.ConditionFalse && etcdRecoveryActiveCondition.Reason == hyperv1beta1.EtcdRecoveryJobFailedReason {
 						etcdManualInterventionRequiredValue := 1.0
 						ch <- prometheus.MustNewConstMetric(
 							etcdManualInterventionRequiredMetricDesc,
 							prometheus.GaugeValue,
 							etcdManualInterventionRequiredValue,
-							append(hclusterLabelValues, metricLabels["rosa_environment"], metricLabels["rosa_id"])...,
+							append(hclusterLabelValues, metricLabels["environment"], metricLabels["internal_id"])...,
+						)
+
+					}
+
+					if sizeOverride := hcluster.Annotations[hyperv1beta1.ClusterSizeOverrideAnnotation]; sizeOverride != "" {
```

Comment:
what happens if the annotation is removed?

Reply:
the metric won't be exposed anymore. So the time series for this HC will stop at the last time the metric showed a value.